### ISSUE TITLE
[java] UnusedAssignment False Positive with Lombok NoArgsConstructor

### ISSUE THREAD

**Issue by esfomeado** [pmd/pmd#5415]
**Affects PMD Version:**  7.8.0

**Rule:** https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedassignment

**Description:** If I have a copy constructor PMD thinks that the initialization is useless

**Code Sample demonstrating the issue:**

```java
@Getter
@NoArgsConstructor
public class Attribute {
    List<String> values = new ArrayList<>();

    public Attribute( Attribute current ) {
        this.values = new ArrayList<>();
    }
}
```

**Expected outcome:**

PMD reports a violation at line 4, but that's wrong. That's a false positive.

**Running PMD through:** Gradle

**Comment by zbynek** [comment]
Looks like a duplicate of https://github.com/pmd/pmd/issues/2817
